boot.s does not compile on arm-none-eabi-gcc v9.2.1

Having recently pulled down this project, I have been unable to compile it on `arm-none-eabi-gcc` v9.2.1 because of a build break caused by this commit:

```
commit 49354ee888a68c1803638a5f87495cc0dfd6fbb4
Author: John Cronin <jncronin@tysos.org>
Date:   Sat Jan 5 16:06:28 2019 +0000

    Add code to ensure a v2 pi is in svc rather than hyp mode
    Thanks to eryjus for highlighting this, see
    https://github.com/jncronin/rpi-boot/issues/23
```


```
gcc -P -traditional-cpp -std=gnu99 -E -o Makefile.rpi-boot -x c Makefile.rpi-boot.in 
make -f Makefile.rpi-boot kernel.img
make[1]: Entering directory '/home/rhett/rpi-boot'
arm-none-eabi-gcc  -include config.h -c boot.s -o boot.o
boot.s: Assembler messages:
boot.s:33: Error: selected processor does not support `cpsid iaf,#0x13' in ARM mode
boot.s:45: Error: Banked registers are not available with this architecture. -- `msr elr_hyp,r3'
boot.s:47: Error: selected processor does not support `eret' in ARM mode
boot.s:72: Error: selected processor does not support `blx r3' in ARM mode
boot.s:75: Error: selected processor does not support `wfe ' in ARM mode
make[1]: *** [Makefile.rpi-boot:173: boot.o] Error 1
make[1]: Leaving directory '/home/rhett/rpi-boot'
make: *** [Makefile:19: kernel.img] Error 2
```

I suspect that the issue is actually just a question of needing appropriate compiler flags, but for the life of me, I cannot determine what those should be.

--- a/Makefile.rpi-boot.in
+++ b/Makefile.rpi-boot.in
@@ -30,6 +30,11 @@ CFLAGS += -mfloat-abi=soft
 CFLAGS += -fno-builtin
 CFLAGS += -fno-tree-loop-distribute-patterns
 
+ASFLAGS += -march=armv7ve
+ASFLAGS += -mcpu=arm1176jzf-s
+ASFLAGS += -mfloat-abi=soft
+
+
 #ifdef DEBUG
 CFLAGS += -g -O0
 ASSERT_OBJS = assert.o

--- a/malloc.c
+++ b/malloc.c
@@ -594,8 +594,13 @@ MAX_RELEASE_CHECK_RATE   default: 4095 unless not HAVE_MMAP
 #define MAX_SIZE_T           (~(size_t)0)
 
 #ifndef USE_LOCKS /* ensure true if spin or recursive locks set */
-#define USE_LOCKS  ((defined(USE_SPIN_LOCKS) && USE_SPIN_LOCKS != 0) || \
-                    (defined(USE_RECURSIVE_LOCKS) && USE_RECURSIVE_LOCKS != 0))
+
+#if (defined(USE_SPIN_LOCKS) && USE_SPIN_LOCKS != 0) || \
+        (defined(USE_RECURSIVE_LOCKS) && USE_RECURSIVE_LOCKS != 0)
+#define USE_LOCKS 1
+#else
+#define USE_LOCKS 0
+#endif
 #endif /* USE_LOCKS */
 
 #if USE_LOCKS /* Spin locks for gcc >= 4.1, older gcc on x86, MSC >= 1310 */

--- a/printf.c
+++ b/printf.c
@@ -194,6 +194,7 @@ reswitch:	switch (ch = (u_char)*fmt++) {
 				padc = '0';
 				goto reswitch;
 			}
+			// fallthrough
 		case '1': case '2': case '3': case '4':
 		case '5': case '6': case '7': case '8': case '9':
 				for (n = 0;; ++fmt) {
@@ -331,6 +332,7 @@ reswitch:	switch (ch = (u_char)*fmt++) {
 			goto handle_nosign;
 		case 'X':
 			upper = 1;
+			// fallthrough
 		case 'x':
 			base = 16;
 			goto handle_nosign;

